app/Models/Post.php

```diff
@@ -277,12 +277,16 @@ public function tag($tag)
 
         $tag = Tag::firstOrCreate(['tag_name' => $tagName], ['tag_slug' => str_slug($tagName, '-')]);
 
-        $this->tags()->attach($tag);
+        // Only tag if the tag doesn't exist on the post yet.
```

Comment:
Shouldn't this toggle tags, so if a post is already tagged with something, when we make the tag request it removes the tag instead of adding (and vice versa)?

Reply:
@sbsmith86 the endpoint [toggles](https://github.com/DoSomething/rogue/blob/master/app/Http/Controllers/TagsController.php#L57-L62) but this [`tag`](https://github.com/DoSomething/rogue/blob/master/app/Models/Post.php#L267) function doesn't actually work like that  - there is another function [`untag`](https://github.com/DoSomething/rogue/blob/master/app/Models/Post.php#L293).

Reply:
ahhh ok 👍